src/Domain/Block.hpp

```diff
@@ -0,0 +1,89 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class template Block.
+
+#pragma once
+
+#include <iosfwd>
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+
+#include "Domain/BlockNeighbor.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/EmbeddingMaps/EmbeddingMap.hpp"
+
+/// \ingroup ComputationalDomain
+/// A Block<VolumeDim> is a region of a VolumeDim-dimensional computational
+/// domain that defines the root node of a tree which is used to construct the
+/// Elements that cover the computational domain.
+///
+/// Each codiminsion 1 boundary of a Block<VolumeDim> is either an external
+/// boundary or identical to a boundary of one other Block.
+///
+/// A Block has logical coordinates that go from -1 to +1 in each
+/// dimension.  The global coordinates are obtained from the logical
+/// coordinates from the EmbeddingMap:  EmbeddingMap::operator() takes
+/// logical coordinates and returns global coordinates.
+///
+/// \tparam VolumeDim the dimension of the Block (i.e. the volume dimension).
+template <size_t VolumeDim>
+class Block {
+  static constexpr int private_asserts() {
+    static_assert(std::is_nothrow_move_constructible<Block>::value and
+                      std::is_nothrow_move_assignable<Block>::value,
+                  "Missing move semantics");
+    return 0;
+  }
+
+ public:
+  /// Construct a Block.
+  ///
+  /// \param embedding_map the EmbeddingMap.
+  /// \param my_id a unique ID.
+  /// \param neighbors info about the Blocks abutting this Block.
```

Comment:
I think it's risky to use words this close in meaning (in my dictionary, "abutting" and "adjacent" are synonyms) to convey different ideas. I like @nilsdeppe's suggestion of stating what we mean using simple words.